server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverConfiguration.java

```diff
@@ -0,0 +1,420 @@
+     */
+    public static RolloverConfiguration parseSetting(String input, String setting) {
+        if (Strings.isNullOrBlank(input)) {
+            throw new IllegalArgumentException("The rollover conditions cannot be null or blank");
+        }
+        String[] sConditions = input.split(",");
+        RolloverConfiguration.ValueParser valueParser = new RolloverConfiguration.ValueParser();
+        for (String sCondition : sConditions) {
+            String[] keyValue = sCondition.split("=");
+            if (keyValue.length != 2) {
+                throw new SettingsException("Invalid condition: '{}', format must be 'condition=value'", sCondition);
+            }
+            var condition = keyValue[0];
+            var value = keyValue[1];
+            if (MaxSizeCondition.NAME.equals(condition)) {
+                valueParser.addMaxIndexSizeCondition(value, setting);
+            } else if (MaxPrimaryShardSizeCondition.NAME.equals(condition)) {
+                valueParser.addMaxPrimaryShardSizeCondition(value, setting);
+            } else if (MaxAgeCondition.NAME.equals(condition)) {
+                valueParser.addMaxIndexAgeCondition(value, setting);
+            } else if (MaxDocsCondition.NAME.equals(condition)) {
+                valueParser.addMaxIndexDocsCondition(value, setting);
+            } else if (MaxPrimaryShardDocsCondition.NAME.equals(condition)) {
+                valueParser.addMaxPrimaryShardDocsCondition(value, setting);
+            } else if (MinSizeCondition.NAME.equals(condition)) {
+                valueParser.addMinIndexSizeCondition(value, setting);
+            } else if (MinPrimaryShardSizeCondition.NAME.equals(condition)) {
+                valueParser.addMinPrimaryShardSizeCondition(value, setting);
+            } else if (MinAgeCondition.NAME.equals(condition)) {
+                valueParser.addMinIndexAgeCondition(value, setting);
+            } else if (MinDocsCondition.NAME.equals(condition)) {
+                valueParser.addMinIndexDocsCondition(value, setting);
+            } else if (MinPrimaryShardDocsCondition.NAME.equals(condition)) {
+                valueParser.addMinPrimaryShardDocsCondition(value, condition);
+            } else {
+                throw new SettingsException("Unknown condition: '{}'", condition);
+            }
+        }
+        return valueParser.getRolloverConfiguration();
+    }
+
+    /**
+     * When max_age is auto we’ll use the following retention dependent heuristics to compute the value of max_age:
+     * - If retention is null aka infinite (default) max_age will be 30 days
+     * - If retention is configured to anything lower than 14 days max_age will be 1 day
+     * - If retention is configured to anything lower than 3 months max_age will be 7 days
+     * - If retention is configured to anything greater than 3 months max_age will be 30 days
+     */
+    private TimeValue evaluateMaxAgeCondition(@Nullable TimeValue retention) {
```

Comment:
Yes, I agree, even I had to pause for a moment to think about the 3 months part, and because month length varies I preferred being more explicit of 90 days. Documentation should reflect that. I will change it.

Reply:
I can also test it directly, if we add more auto conditions in the future it will be easier to test.